### DISCUSSION TITLE
为什么查询命中rollup之后查询反而变慢了

### DISCUSSION THREAD

**Discussion by LangYuanzh** [apache/doris#8024] [A - General / Q&A]
Doris 版本0.15
查询一张Uniq表 table_test，总共40个字段，其中UNIQUE KEY为(`a(int)`, `b(int)`, `c(int)`, `d(date)`, `e(varchar)`, `f(varchar)`);
数据量：25000条数据
查询语句： select xxx from table_test where d = '2021-01-01' and e in (... 总共78个字段); 查询结果为78条数据。
1. 不添加任何rollup: 查询耗时0.05秒
2. 添加rollup更改key顺序为(`d(date)`, `e(varchar)`,`a(int)`, `b(int)`, `c(int)`,  `f(varchar)`)，查询执行计划命中该rollup，但是查询时间增加至0.2秒多。

**Comment by PRNDA** [comment]
数据量太少的时候，rollup意义不大。
就像数据库索引一样，有时候虽然你建有索引，但是引擎评估发现全表扫描反而更快的时候，它就不走你索引，很正常的。

**Comment by ChPi** [comment]
是多执行几次的结果吗？ 看是不是缓存的原因

**Comment by qzsee** [comment]
看一下profile
